tests/fuzzer/fuzzer_stdcm_single_timetable.py

```diff
@@ -0,0 +1,199 @@
+import datetime
+import json
+import random
+from dataclasses import dataclass
+from pathlib import Path
+from typing import Dict, Iterable, List, Optional
+
+import requests
+
+from fuzzer import (
+    _get_random_rolling_stock,
+    _random_set_element,
+    _to_ms,
+    get_infra,
+)
+from tests.scenario import Scenario
+
+_TIMEOUT = 300
+
+_EDITOAST_URL = "http://127.0.0.1:8090/"
+_INFRA_NAME = "France"
+_TIMETABLE_ID = 0
+
+"""
+Generates random stdcm requests on a single infra + fixed timetable.
+Much lighter than `fuzzer.py`, but can't generate regression tests.
+Inputs that can't directly be set in the (non-debug) STDCM UI are fixed.
+
+This isn't automatically run anywhere, it's only used by hand for more convenient
+testing. May not always be up-to-date.
+
+Note: on an imported infra and timetables, tests take a while.
+"""
+
+
+@dataclass
+class STDCMException(Exception):
+    error: str | Dict
+    status_code: Optional[int] = None
+    payload: Optional[Dict] = None
+
+
+@dataclass
+class TimetableTimeRange:
+    start: datetime.datetime
+    end: datetime.datetime
+
+    def make_random_time(self) -> str:
+        delta = (self.end - self.start).seconds
+        date = self.start + datetime.timedelta(seconds=(random.randint(0, delta)))
+        return date.isoformat()
+
+
+def run(
+    editoast_url: str,
+    scenario: Scenario,
+    n_test: int = 1000,
+    log_folder: Optional[Path] = None,
+    seed: Optional[int] = None,
```

Comment:
nit: should we have a default seed for reproducibility? or make it mandatory

Reply:
Reproducibility isn't a goal here, I can't really think of a use case where we'd need it. Other than debugging the script itself, but the optional parameter should be enough.

Reply:
Yes, when we receive a 500 response, the content (detailed error) is put in an exception. It's then caught and logged in a file, alongside the payload used to reproduce it.

Reply:
I see, thanks.